src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionListener.cs

```diff
@@ -171,12 +174,6 @@ public ValueTask DisposeAsync()
             // Dispose the memory pool
             _memoryPool.Dispose();
 
-            // Dispose any pooled senders
-            foreach (var setting in _settings)
```

Comment:
Where did this go?

Reply:
I moved this into the dispose of the new factory: https://github.com/dotnet/aspnetcore/blob/7465741d00784841555dbb5a59671d8b414340dc/src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionContextFactory.cs#L63

The pooled sender/settings and transport schedule bits definitely feel a bit wrong still

Reply:
I think we need to add a singleton one to the container but we can't use it in the transport. We need to create one per listener in the transport.